_ml-commons-plugin/gpu-acceleration.md

```diff
@@ -0,0 +1,224 @@
+---
+layout: default
+title: GPU acceleration
+parent: Model-serving framework
+nav_order: 150
+---
+
+# GPU acceleration 
+
+When running a natural language processing (NLP) model in your OpenSearch cluster with a machine learning (ML) node, you can achieve better performance on the ML node using graphic processing unit (GPU) acceleration. GPUs can work in tandem with the central processing unit (CPU) memory of your cluster to speed up the model upload and training. 
+
+## Supported GPUs
+
+Currently, ML nodes only support GPU acceleration from two GPU devices:
+
+- [NVIDIA GPU](https://aws.amazon.com/nvidia/)
+- [AWS Inferentia](https://aws.amazon.com/machine-learning/inferentia/)
+
+If you need GPU power, you can provision GPU instances through [Amazon EC2](https://aws.amazon.com/ec2/). For more information on how to provision a GPU instance, see [Recommended GPU Instances](https://docs.aws.amazon.com/dlami/latest/devguide/gpu.html).
+
+## Supported images
+
+GPU acceleration supports both Docker images and [Amazon Machine Images (AMI)](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html).
+
+## Setting up a GPU accelerated ML node
+
+Depending on the GPU, you can provision an GPU accelerated ML node using automated initialization scripts or manually. 
+
+### Mount an nvidia-uvm device
+
+NVIDIA uses CUDA®, NVIDIA's programming platform, to increase node performance. In order to take advantage of CUDA, you need to make sure that your drivers contain the `nvidia-uvm` kernels inside the `/dev` directory. To check for the kernels, enter `ls -al /dev | grep nvidia-uvm`.
+
+If the `nvidia-uvm` does not exist, run the `nvidia-uvm-init.sh` script. 
+
+```
+#!/bin/bash
+## Script to initialize nvidia device nodes.
+## https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#runfile-verifications
+/sbin/modprobe nvidia
+if [ "$?" -eq 0 ]; then
+  # Count the number of NVIDIA controllers found.
+  NVDEVS=`lspci | grep -i NVIDIA`
+  N3D=`echo "$NVDEVS" | grep "3D controller" | wc -l`
+  NVGA=`echo "$NVDEVS" | grep "VGA compatible controller" | wc -l`
+  N=`expr $N3D + $NVGA - 1`
+  for i in `seq 0 $N`; do
+    mknod -m 666 /dev/nvidia$i c 195 $i
+  done
+  mknod -m 666 /dev/nvidiactl c 195 255
+else
+  exit 1
+fi
+/sbin/modprobe nvidia-uvm
+if [ "$?" -eq 0 ]; then
+  # Find out the major device number used by the nvidia-uvm driver
+  D=`grep nvidia-uvm /proc/devices | awk '{print $1}'`
+  mknod -m 666 /dev/nvidia-uvm c $D 0
+  mknod -m 666 /dev/nvidia-uvm-tools c $D 0
+else
+  exit 1
+fi
+```
+
+After verification that `nvidia-uvm` exists under `/dev`, you can start OpenSearch inside your cluster. 
+
+### Mount AWS Inferentia
```

Comment:
This is not to "Mount" AWS Inferentia, just steps to prepare AWS Inferentia ML node. For AWS Inferentia , it doesn't have device mount issue.

Reply:
Switched to "Prepare".

Reply:
Prepare looks good, then maybe we should change to "Prepare AWS Inferentia ML node"